services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -69,6 +72,37 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                     app=app,
                 )
 
+                await remove_resources_if_guest_user(
+                    app=app,
+                    project_uuid=resource_value,
+                    user_id=int(key['user_id'])
+                )
+
+
+async def remove_resources_if_guest_user(
+    app: web.Application,
+    project_uuid: str,
+    user_id: int
+):
+    """When a guest user finishes using the platform its Posgtres
+    and S3/MinIO entries need to be removed
+    """
+    logger.debug("Will try to remove resoruces if guest for user id '%s'", user_id)
+    db = get_storage(app)
+    user = await db.get_user({"id": user_id})
+    logger.debug("Detected user %s", user)
+    if not user:
+        logger.debug("Could not find the user with id %s, skipping", user_id)
+        return
+
+    logger.debug("Found user '%s' with role '%s'", user["email"], user["role"])
+    if UserRole(user["role"]) != UserRole.GUEST:
+        return
+
+    logger.info("Removing guest user's '%s' db entries.", user["email"])
+    await delete_project_from_db(app, project_uuid, user_id)
+    await db.delete_user(user)
```

Comment:
I actually created a users_api.py file in the [PR in here](https://github.com/ITISFoundation/osparc-simcore/pull/1310)
maybe it might be nice to have a function to get the user role from the id. somethign like ```is_user_guest(user_id)->bool```
Then I would wonder what should happen if you don't find the user in the DB. that means either it was already deleted or changed. You should probably at least issue a warning if that is the case. And what about the project data? it should still be removed don't you think?

Reply:
I will also create a ** users_api.py** file in the same place with similar structure. We shall merge them later.